### ISSUE TITLE
F841 false positive in match case with constant

### ISSUE THREAD

**Issue by ember91** [astral-sh/ruff#16232]
### Description

List of keywords searched before creating this issue:

  - `F841`
 
Minimal code snippet that reproduces the bug:

```python
# test.py
C = 13

def f():
    match 13:
        case C:
            pass
```

Invoked command:

```bash
ruff check --isolated --select F841 test.py
```

Current Ruff settings: -

Current Ruff version: `ruff 0.9.6`

I expected it to not mark line 6 with `F841`.

**Comment by ntBre** [comment]
I think this is a true positive because Python doesn't treat the `C` in the `case` as a constant:

```console
>>> C = 13
>>> match 2:
...     case C:
...         print(C)
...
2
```

It should actually probably mark both `C = 13` *and* `case C` as unused.

I believe you have to use an [enum](https://peps.python.org/pep-0636/#matching-against-constants-and-enums) (or other dotted name) to match against a constant like this.

**Comment by ember91** [comment]
Wait, is it pattern matching my constant?

If so I guess the behavior's correct. It's defining a variable and it's unused. Just a bit confusing.

I think you can close this one.

**Comment by ntBre** [comment]
Oops, I forgot that F841 only applies to function scopes, so it obviously won't catch the outer `C`. I'll go ahead and close this. Thanks!

**Comment by ntBre** [comment]
I took a closer look at this, and this *is* a bug that we'd like to fix with the Rust-style behavior, but it's being tracked in https://github.com/astral-sh/ruff/issues/6704, so I *will* close this for now. Thanks for opening the issue!
